lib/alg.ml

```diff
@@ -520,17 +509,97 @@ module Alg2T (Alg : Alg_t) : Alg_t = struct
   (*                ^^^^^^^^^^^^^^^^^^
                      The only change!
   *)
-  let f_tilde = Topo.lift_tilde f
+  let f_tilde = Topo.lift_tilde f Alg.topology
 
   let z' s pkt =
-    let pt, s' = Alg.control.z s pkt in
-    (f_tilde Alg.topology pt, s')
+    let pt, s', ts = Alg.control.z s pkt in
+    (f_tilde pt, s', ts)
 
   let control : Control.t =
-    { s = Alg.control.s; q = Pifotree.create topology; z = z' }
+    { s = Alg.control.s; q = Pieotree.create topology; z = z' }
```

Comment:
Nice, I'm thrilled that the changes in this file up to now were relatively straightforward. Thanks for tidying up my silly string-based matches into a little type of its own ("A" -> A), that's much nicer